blocks/bricks/interfaces.py

```diff
@@ -127,45 +128,97 @@ class Initializable(RNGMixin, Brick):
         ``True``.
     use_bias : :obj:`bool`, optional
         Whether to use a bias. Defaults to `True`. Required by
-        :meth:`~.Brick.initialize`. Only supported by bricks for which
-        :attr:`has_biases` is ``True``.
+        :meth:`~.Brick.initialize`.
     rng : :class:`numpy.random.RandomState`
 
-    Attributes
-    ----------
-    has_biases : bool
-        ``False`` if the brick does not support biases, and only has
-        :attr:`weights_init`.  For an example of this, see
-        :class:`.Bidirectional`. If this is ``False``, the brick does not
-        support the arguments ``biases_init`` or ``use_bias``.
-
     """
-    has_biases = True
+
+    initializable_roles = [WEIGHT, BIAS, FILTER, INITIAL_STATE]
 
     @lazy()
-    def __init__(self, weights_init=None, biases_init=None, use_bias=None,
+    def __init__(self, initialization_schemes=None, use_bias=True,
                  seed=None, **kwargs):
-        super(Initializable, self).__init__(**kwargs)
-        self.weights_init = weights_init
-        if self.has_biases:
-            self.biases_init = biases_init
-        elif biases_init is not None or not use_bias:
-            raise ValueError("This brick does not support biases config")
-        if use_bias is not None:
-            self.use_bias = use_bias
+        self.use_bias = use_bias
         self.seed = seed
+        self.initialization_schemes = initialization_schemes
+        self.parameter_roles = set([])
+        if self.initialization_schemes is None:
+            self.initialization_schemes = {}
+
+        initialization_to_role = {"weights_init": WEIGHT, 'biases_init': BIAS,
```

Comment:
For the record, it looks like you have chosen to only handle a handful of `<role>_init` keyword arguments, not all of them. This is probably good, because this clearly makes `initialization_schemes` the primary way of defining the initialization schemes.

But the code clearly lacks a check `if key not in initialization_to_role`. Since now we have a clearly defined white list, I would probably raise a warning, that calling the arguments `smth_init` is recommended, but still pass this argument to the `super().__init__` call.